### PR TITLE
Connect plans table

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- Use the `pay` flag with guiproxy on `staging`.
- open `deployment-flow.js`
- add {this._generatePricingSection()} to the `render` method after `{this._generateModelNameSection()}
- load the gui
- add `cs:~cmarstech/mattermost` and open the deployment flow
- you should see the plans table and details of the plan for mattermost

**Comment by huwshimi** [comment]
[Hook up plans table](https://trello.com/c/Jof0UGrf/538-hook-up-plans-table)

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8646/

**Comment by frankban** [comment]
Ignoring errors due to terms that cannot be found I can see plans in the page, so QA is ok.

screen shot 2018-04-24 at 13 30 16

File: jujugui/static/gui/src/app/components/deployment-flow/pricing/plan-table/plan-table.js [hunk 1]
```diff
@@ -1,80 +1,101 @@
...
+  }
+
+  /**
+    Callback for when plans for an entity have been successfully fetched.
+    @param charmId {String} The charm id for the returned plans.
+    @param error {String} An error message, or null if there's no error.
+    @param plans {Array} A list of the plans found.
+  */
+  _getPlansCallback(charmId, error, plans=[]) {
+    const newPlans = this.state.plans;
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/deployment-flow/pricing/plan-table/plan-table.js]: We should check for errors here before setting plans to state.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8648/

**Comment by huwshimi** [comment]
Thanks @frankban, error handling added :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**